src/services/PromotionService.ts

```diff
@@ -0,0 +1,113 @@
+import { FilterOptions, Promotion, Sort } from '../types/promotion';
+import Routes from '../utils/routes';
+
+type PromotionsSuccess = Promotion[];
+
+type PromotionsError = {
+  errorCode: string;
+  message: string[];
+};
+
+type PromotionsResponse = PromotionsSuccess | PromotionsError;
+
+/**
+ * Fetches entire list of promotions. If a query object is given, filters the promotions according to the given query.
+ * If an error occurs, an empty list will be returned.
+ *
+ * @param query [optional] - An array of objects with key-value pairs for the query parameters
+ */
+export async function getPromotions(query?: Record<string, string>[]): Promise<Promotion[]> {
```

Comment:
ALso is there some way we can make this typed rather than a `Record`? For example an interface like `PromotionQuery`? 

E.g. in BE we have this DTO for queries
https://github.com/ubclaunchpad/PromoPal-backend/blob/3c50736cd1a8065bdbf8711b05c990e2f18f93ad/src/main/validation/PromotionQueryValidation.ts#L39

Reply:
Ahh good point, yeah I can change this!